client/virtual_channel_test.go

```diff
@@ -68,6 +68,60 @@ func TestVirtualChannelsOptimistic(t *testing.T) {
 	assert.NoError(t, err, "Bob: invalid final balances")
 }
 
+func TestVirtualChannelsDispute(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), testDuration)
+	defer cancel()
+
+	vct := setupVirtualChannelTest(t, ctx)
+	var err error
+	assert := assert.New(t)
+
+	time.Sleep(100 * time.Millisecond) // Sleep to ensure that virtual channel watcher is running.
+
+	// Alice registers and settles channel with Ingrid.
```

Comment:
I will think of something better here, but for now its faster if we just keep the discussion open and focus on the more important things i guess.

Reply:
This should be able to replace `TestVirtualChannelsDispute` + `TestVirtualChannelsDisputeHubFirst`.  
It did not work parallel because the test setup seems to race.  

```go
func TestVirtualChannelsDispute(t *testing.T) {
	ctx, cancel := context.WithTimeout(context.Background(), testDuration)
	defer cancel()

	vct := setupVirtualChannelTest(t, ctx)
	time.Sleep(100 * time.Millisecond) // Sleep to ensure that virtual channel watcher is running.

	chs := []*client.Channel{vct.chAliceIngrid, vct.chIngridAlice, vct.chBobIngrid, vct.chIngridBob}
	// Close the channels in a random order.
	for _, i := range rand.Perm(len(chs)) {
		err := vct.registerAndSettle(ctx, chs[i])
		assert.NoError(t, err, "alice-ingrid: register and settle")
	}
}
```

Sorry for just pasting it in here, but i dont like to push to other peoples WIP branches.  
Btw is it intentional that the channels `AliceBob` and `BobAlice` are not closed?

Reply:
Very nice idea and thanks for the code receipt.